### PR TITLE
Revise s_client and s_server verbiage re secure renegotiation.

### PR DESCRIPTION
**Pull request by FGasper**
Since TLS v1.3 eschews renegotiation entirely it’s misleading to have
these apps say it’s “not supported” when in fact the TLS version is
new enough not to need renegotiation at all.

##### Checklist
- [x] documentation is added or updated

### CODE REVIEW

**Comment by beldmit** [comment]
close/reopen to kick CLA bot

File: CHANGES.md [hunk 1]
```diff
@@ -24,6 +24,11 @@ OpenSSL 3.1
 
 ### Changes between 3.0 and 3.1 [xx XXX xxxx]
 
+ * s_client and s_server apps now explicitly say when the TLS version
+   obviates a need for renegotiation. This avoids confusion between
```
Comment (kaduk) [inline CHANGES.md:28]:
```suggestion
   does not include the renegotiation mechanism. This avoids confusion between
```

File: apps/lib/s_cb.c [hunk 2]
```diff
@@ -1543,3 +1546,13 @@ void print_ca_names(BIO *bio, SSL *s)
         BIO_write(bio, "\n", 1);
     }
 }
+
+void ssl_print_secure_renegotiation_notes(BIO* bio, SSL* s)
```
Comment (kaduk) [inline apps/lib/s_cb.c:1550]:
```suggestion
void ssl_print_secure_renegotiation_notes(BIO *bio, SSL *s)
```

**Review by kaduk — COMMENTED** [review]
Looks good modulo the inline nits/suggestions.  (And thank you for updating the top-of-file comment in `s_cb.c`!)

I will also close/reopen to trigger the CLA check again.

File: apps/lib/s_cb.c (hunk 2 above)
Comment (FGasper) [reply apps/lib/s_cb.c:1550]: done

File: CHANGES.md (hunk 1 above)
Comment (FGasper) [reply CHANGES.md]: done

**Review by paulidale — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

File: CHANGES.md [hunk 3]
```diff
@@ -24,6 +24,11 @@ OpenSSL 3.1
 
 ### Changes between 3.0 and 3.1 [xx XXX xxxx]
 
+ * s_client and s_server apps now explicitly say when the TLS version
+   does not include the renegotiation mechanism. This avoids confusion
+   between that scenario versus when the TLS version includes secure
+   renegotiation but the peer lacks support for it.
+
```
Comment (t8m) [inline CHANGES.md:31]: Please add your name here similarly to other entries.

**Review by t8m — APPROVED** [review]
Just one nit.

File: CHANGES.md (hunk 3 above)
Comment (FGasper) [reply CHANGES.md:31]: Done.

**Review by kaduk — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]
Reconfirm

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Squashed and merged to master branch. Thank you for your contribution.

**Pull request closed without merging**